dbt/adapters/spark/pysparkcon.py

```diff
@@ -0,0 +1,93 @@
+
+from __future__ import annotations
+
+import datetime as dt
+from types import TracebackType
+from typing import Any
+
+from dbt.events import AdapterLogger
+from dbt.utils import DECIMALS
+
+
+from pyspark.rdd import _load_from_socket
+import pyspark.sql.functions as F
+
+
+import importlib
+import sqlalchemy
+import re
+
+logger = AdapterLogger("Spark")
+NUMBERS = DECIMALS + (int, float)
+
+
+class PysparkConnectionWrapper(object):
+    """Wrap a Spark context"""
+
+    def __init__(self, python_module):
+        self.result = None
+        if python_module:
```

Comment:
I prefer to avoid such a hook, it's very specific. The `python_module` is a unexpected parameter for `PysparkConnectionWrapper`, it's unclear why it is needed and how it works. 

We could add docs about this, still it is confusing to write `PysparkConnectionWrapper(python_module)`

Reply:
I can change it, what do you propose ?